### PR TITLE
Fix Brave content settings page crash

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/17808

1. `ContentSettingsTypes` is no longer accessible directly in site_details.js and privacy_pages.js, we have to get the type through `contentSettingsTypesEnum_` due to https://chromium-review.googlesource.com/c/chromium/src/+/2997906
2. `AUTOPLAY` should be added to `VisiblePermissionCategories` so it can be shown in site details page, related upstream change https://chromium-review.googlesource.com/c/chromium/src/+/2915454
3. Eliminate some `AUTOPLAY` polymer patches
4. Add Ethereum request to site details
Screen Shot 2021-09-07 at 19 25 34

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

#### Crash
1. Go to brave://settings/content/autoplay
2. Brave should not crash
#### Site details
1. Go to brave://settings/content/siteDetails?site=https%3A%2F%2Fbrave.com
2. Autoplay should be in permission list

#### Crash
1. Go brave://flags/#native-brave-wallet and enable native wallet
2. Go to brave://settings/content/ethereum
3. Brave should not crash
#### Site details
1. Go to brave://settings/content/siteDetails?site=https%3A%2F%2Fbrave.com
2. There is no Ethereum permission in the list
3. Go brave://flags/#native-brave-wallet and enable native wallet
4. Go to brave://settings/content/siteDetails?site=https%3A%2F%2Fbrave.com
5. Ethereum should be in permission list

### CODE REVIEW

**Review by yrliou — APPROVED** [review]
LGTM

File: browser/resources/settings/brave_overrides/privacy_page.js [hunk 1]
```diff
@@ -12,18 +12,46 @@ RegisterPolymerTemplateModifications({
...
+                category="[[contentSettingsTypesEnum_.AUTOPLAY]]"
+                block-header="${I18nBehavior.i18n('siteSettingsBlock')}">
+            </category-setting-exceptions>
+          </settings-subpage>
+        </template>
+      `)
       const isNativeBraveWalletEnabled = loadTimeData.getBoolean('isNativeBraveWalletFeatureEnabled')
       if (isNativeBraveWalletEnabled) {
         pages.insertAdjacentHTML('beforeend', `
+          <template is="dom-if" route-path="/content/autoplay" no-search>
```
Comment (mkarolin) [inline browser/resources/settings/brave_overrides/privacy_page.js:33]: Not sure I am following why autoplay is duplicated here under `isNativeBraveWalletEnabled`

Comment (darkdh) [reply browser/resources/settings/brave_overrides/privacy_page.js]: fixed, thanks

**Review by mkarolin — APPROVED** [review]
LGTM

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.31.29 Chromium: 93.0.4577.63 (Official Build) nightly (64-bit)
-- | --
Revision | ff5c0da2ec0adeaed5550e6c7e98417dac77d98a-refs/branch-heads/4577@{#1135}
OS | Windows 10 OS Version 2009 (Build 19042.1165)
```

### Autoplay Test Cases

#### **`Crash`**

* opened `brave://settings/content/autoplay` and ensured that Brave didn't crash (waiting ~30s as well)

#### **`Site details`**

autoplaySettings

### Ethereum request

#### **`Crash`**

* enabled native ethereum wallet via `brave://flags/#native-brave-wallet` and restarted brave
* opened `brave://settings/content/ethereum` and ensured that Brave didn't crash (waiting ~30s as well)

#### **`Site details`**

**`Example Disabled`** | **`Example Enabled`**
-------------------------|---------------------------
noETH | ethAvailable

**Pull request merged**